lightning-c-bindings/src/lightning_background_processor.rs

```diff
@@ -0,0 +1,157 @@
+		let mut ret = (self.persist_manager)(self.this_arg, &crate::lightning::ln::channelmanager::ChannelManager { inner: unsafe { (channel_manager as *const _) as *mut _ }, is_owned: false });
+		let mut local_ret = match ret.result_ok { true => Ok( { () /*(*unsafe { Box::from_raw(<*mut _>::take_ptr(&mut ret.contents.result)) })*/ }), false => Err( { (*unsafe { Box::from_raw(<*mut _>::take_ptr(&mut ret.contents.err)) }).to_rust() })};
+		local_ret
+	}
+}
+
+// We're essentially a pointer already, or at least a set of pointers, so allow us to be used
+// directly as a Deref trait in higher-level structs:
+impl std::ops::Deref for ChannelManagerPersister {
+	type Target = Self;
+	fn deref(&self) -> &Self {
+		self
+	}
+}
+/// Calls the free function if one is set
+#[no_mangle]
+pub extern "C" fn ChannelManagerPersister_free(this_ptr: ChannelManagerPersister) { }
+impl Drop for ChannelManagerPersister {
+	fn drop(&mut self) {
+		if let Some(f) = self.free {
+			f(self.this_arg);
+		}
+	}
+}
+/// Start a background thread that takes care of responsibilities enumerated in the top-level
+/// documentation.
+///
+/// If `persist_manager` returns an error, then this thread will return said error (and
+/// `start()` will need to be called again to restart the `BackgroundProcessor`). Users should
+/// wait on [`thread_handle`]'s `join()` method to be able to tell if and when an error is
+/// returned, or implement `persist_manager` such that an error is never returned to the
+/// `BackgroundProcessor`
+///
+/// `persist_manager` is responsible for writing out the [`ChannelManager`] to disk, and/or
+/// uploading to one or more backup services. See [`ChannelManager::write`] for writing out a
+/// [`ChannelManager`]. See [`FilesystemPersister::persist_manager`] for Rust-Lightning's
+/// provided implementation.
+///
+/// [`thread_handle`]: BackgroundProcessor::thread_handle
+/// [`ChannelManager`]: lightning::ln::channelmanager::ChannelManager
+/// [`ChannelManager::write`]: lightning::ln::channelmanager::ChannelManager#impl-Writeable
+/// [`FilesystemPersister::persist_manager`]: lightning_persister::FilesystemPersister::persist_manager
+#[must_use]
+#[no_mangle]
+pub extern "C" fn BackgroundProcessor_start(mut persister: crate::lightning_background_processor::ChannelManagerPersister, mut event_handler: crate::lightning::util::events::EventHandler, chain_monitor: &crate::lightning::chain::chainmonitor::ChainMonitor, channel_manager: &crate::lightning::ln::channelmanager::ChannelManager, peer_manager: &crate::lightning::ln::peer_handler::PeerManager, mut logger: crate::lightning::util::logger::Logger) -> BackgroundProcessor {
```

Comment:
Yay super excited for this to be exposed! I wonder how this is gonna reconcile with e.g. `ChannelManagerPersister` and other Java sample modules that currently are redundant to some of `BackgroundProcessor`'s job?

Reply:
My plan was to drop the similar Java wrappers entirely and move towards common wrappers, but still working on making it work sensibly with `stop` taking `self` ownership.